src/components/Accordion/Accordion.ts

```diff
@@ -42,39 +53,67 @@ export class CandyAccordion extends LitElement {
     }
   };
 
+  doesComponentHasIcon = () =>
+    Array.from(this.children).some((child: Element) => child.slot === "icon");
+
+  onSlotIconChange = () => {
+    this.hasSlotIcon = this.doesComponentHasIcon();
+  };
+
   render() {
-    const styleClass = this.active ? "text-blue" : "text-black";
+    const textColor = this.active ? "text-blue" : "text-black";
     const slotContent = html`
       <div class="accordion-childrens ${this.collapsed || !this.active ? "hidden" : null}">
         <slot></slot>
       </div>
     `;
 
+    const renderOptionsIcons =
+      this.minimizeOptions && this.countOptionsSlotAmount() >= 2
+        ? html` <div class="end-icons">
+            <candy-popover
+              class="options-container"
+              ?isActive=${this.isPopoverActive}
+              side=${this.optionsPopoverSide}
+            >
+              <img
+                style="width: 24px; height: 24px; color: pink !important;"
+                src=${this.active ? VerticalMenuBlue : VerticalMenuBlack}
+                alt="accordion-vertical-icon"
+                data-testid="accordion-options-icon"
+              />
+              <div slot="content" class="options-container">
+                <slot name="options"></slot>
+              </div>
+            </candy-popover>
+          </div>`
+        : html`<div class="end-icons">
+            <div class="options-container">
+              <slot name="options"></slot>
+            </div>
+          </div> `;
+
     return html`
       <div part="accordion">
         <button
           part="accordion-button"
           role="button"
           href="#"
-          class="${"element-container " + styleClass} ${!this.collapsed
+          class="${"element-container " + textColor} ${!this.collapsed
             ? "element-container-extended"
             : null}"
           ?disabled="${this.disabled}"
           @click="${this.handleClick}"
         >
-          <slot name="icon"></slot>
-          <p part="accordion-text">${!this.collapsed ? this.label : null}</p>
-          ${!this.collapsed
-            ? html`<div class="end-icons">
-                  <div class="options-container">
-                    <slot name="options"></slot>
-                  </div>
-                </div>
-                <div
-                  part="accordion-chevron"
-                  class=${`chevron ${this.active ? "rotate" : ""}`}
-                ></div> `
-            : null}
+          <slot @slotchange=${this.onSlotIconChange} name="icon"></slot>
+          <p part="accordion-text" class="${this.collapsed ? "display-none" : ""}">${this.label}</p>
+          ${!this.collapsed ? renderOptionsIcons : null}
+          ${this.collapsed && this.hasSlotIcon
+            ? null
+            : html`<div
+                part="accordion-chevron"
+                class=${`chevron ${this.active ? "rotate" : ""}`}
+              ></div>`}
```

Comment:
regarder si possible d'organiser le rendu avec un v-if else comme suit:

```html
if (this.collapsed)
<p part="accordion-text" class="display-none">${this.label}</p>
          ${this.hasSlotIcon
            ? null
            : html`<div
                part="accordion-chevron"
                class=${`chevron ${this.active ? "rotate" : ""}`}
              ></div>`}
else

<p part="accordion-text">${this.label}</p>
${renderOptionsIcons }
```

Reply:
@AntonyDerache besoin que tu ajoutes un commentaire pour comprendre ce que tu as pu faire ou constater

Reply:
Pourtant j'ai bien effectué les changements. Voici le code en question dans accordion.ts

```
${this.collapsed
    ? html` ${this.hasSlotIcon
        ? null
        : html`<div
            part="accordion-chevron"
            class=${`chevron ${this.active ? "rotate" : ""}`}
          ></div>`}`
    :  html`
       <p part="accordion-text">${this.label}</p>
       ${renderOptionsIcons}
       <div
            part="accordion-chevron"
            class=${`chevron ${this.active ? "rotate" : ""}`}
        ></div>
`}`

Reply:
ok vu. je pense que c'est parce que la syntaxe me semble encore trop compliquée / trop d'utilisation de `this.collapsed ? x : y`
On verra ça dans un deuxième temps  sur l'ensemble de la lib